community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/SchemaRuleSerialization.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store.record;
+
+import java.nio.ByteBuffer;
+
+import org.neo4j.kernel.api.exceptions.schema.MalformedSchemaRuleException;
+import org.neo4j.kernel.api.index.SchemaIndexProvider;
+import org.neo4j.kernel.api.schema_new.LabelSchemaDescriptor;
+import org.neo4j.kernel.api.schema_new.RelationTypeSchemaDescriptor;
+import org.neo4j.kernel.api.schema_new.SchemaDescriptor;
+import org.neo4j.kernel.api.schema_new.SchemaDescriptorFactory;
+import org.neo4j.kernel.api.schema_new.SchemaProcessor;
+import org.neo4j.kernel.api.schema_new.index.NewIndexDescriptor;
+import org.neo4j.kernel.api.schema_new.index.NewIndexDescriptorFactory;
+import org.neo4j.storageengine.api.schema.SchemaRule;
+import org.neo4j.string.UTF8;
+
+import static java.lang.String.format;
+import static org.neo4j.string.UTF8.getDecodedStringFrom;
+
+public class SchemaRuleSerialization
+{
+    // Schema rule type
+    private final static byte INDEX_RULE = 1, CONSTRAINT_RULE = 2;
+
+    // Index type
+    private final static byte GENERAL_INDEX = 51, UNIQUE_INDEX = 52;
+
+    // Schema type
+    private final static byte SIMPLE_LABEL = 101, SIMPLE_REL_TYPE = 102;
```

Comment:
Hmm I tried to make this work, but it became a lot of code for some type safety while loosing readability. It's not so easy to write compact enums in java if they should have fields. Anyway: put that into a separate branch if you want to push that direction https://github.com/fickludd/neo4j/blob/3.2-new-schema-store-with-enums/community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/SchemaRuleSerialization.java